swarm/pss/pss.go

```diff
@@ -699,23 +710,19 @@ func (self *Pss) forward(msg *PssMsg) {
 		}
 
 		// get the protocol peer from the forwarding peer cache
-		sendMsg := fmt.Sprintf("MSG %x TO %x FROM %x VIA %x", digest, to, self.BaseAddr(), op.Address())
+		sendMsg := fmt.Sprintf("MSG TO %x FROM %x VIA %x", to, self.BaseAddr(), op.Address())
 		self.fwdPoolMu.RLock()
 		pp := self.fwdPool[sp.Info().ID]
 		self.fwdPoolMu.RUnlock()
-		if self.checkFwdCache(op.Address(), digest) {
-			log.Trace(fmt.Sprintf("%v: peer already forwarded to", sendMsg))
+
+		// attempt to send the message
```

Comment:
The `pp.Send(msg)` used to be asynchronous, and now is synchronous. I think this will deadlock on the `sim` adapter. Any reason the go routine was removed?

Reply:
Yes, because we want to catch the error.

The loop doing queue processing is in a go routine in itself now, separate from the other handler code. Maybe it makes a difference. Let's see how it behaves with your refactored tests.

Reply:
Oh right - then it might be just fine as is.

Reply:
We will see :)

Reply:
I rebased and all tests passed :)